examples/step-60/doc/intro.dox

```diff
@@ -0,0 +1,195 @@
+$H^1(\Gamma)$ when the intrinsic dimension of $\Gamma$ is the same of $\Omega$,
+and that the same is true, with a less regular range space (namely
+$H^{1/2}(\Gamma)$), when the dimension of $\Gamma$ is one less with respect to
+$\Omega$, and $\Gamma$ does not have a boundary. In this second case, the
+operator $\gamma$ is also known as the *trace* operator, and it is well defined
+for Lipschitz co-dimension one curves and surfaces $\Gamma$ embedded in
+$\Omega$.
+
+The co-dimension two case is a little more complicated, and in general it is
+not possible to construct a continous trace operator, not even from
+$H^1(\Omega)$ to $L^2(\Gamma)$, when the dimension of $\Gamma$ is zero or one
+respectively in two and three dimensions.
+
+In this tutorial program we're not interested into further details on $\gamma$:
+we give $\gamma$'s extension for granted assuming that the dimension of the
+embedded domain (`dim`) is always smaller by one or equal w.r.t. the dimension
+of the embedding domain $\Omega$ (`spacedim`).
+
+We are going to solve the following differential problem: given a sufficiently
+regular function $g$ on $\Gamma$, find the solution $u$ to
+
+@f{eqnarray*}
+- \Delta u + \gamma^T \lambda &=& 0 & \text{ in } \Omega\\
+\gamma u &=& g &  \text{ in } \Gamma \\
+u & = & 0 & \text{ on } \partial\Omega.
+@f}
+
+This is a constrained problem, where we are looking for a harmonic function
+$u$, that satisfies homogeneous boundary conditions on $\partial\Omega$, subject
+to the constraint $\gamma u = g$ using a Lagrange multiplier.
+
+The variational formulation can be derived by introducing two infinite
+dimensional spaces $V(\Omega)$ and $Q^*(\Gamma)$, respectively for the solution
+$u$ and for the Lagrange multiplier $\lambda$.
+
+Multiplying the first equation by $v \in V(\Omega)$ and the second by $q \in
+Q(\Gamma)$, integrating by parts when possible, and exploiting the boundary
+conditions on $\partial\Omega$, we obtain the following variational problem:
+
+Given a sufficiently regular function $g$ on $\Gamma$, find the solution $u$ to
+@f{eqnarray*}
+(\nabla u, \nabla v)_{\Omega} + (\lambda, \gamma v)_{\Gamma} &=& 0 & \forall v \in V(\Omega) \\
+(\gamma u, q)_{\Gamma} &=& (g,q)_{\Gamma} &  \forall q \in Q(\Gamma),
+@f}
+
+where $(\cdot, \cdot)_{\Omega}$ (respectively $(\cdot, \cdot)_{\Gamma}$)
+represent the $L^2$ scalar product in $\Omega$ (respectively in $\Gamma$).
+
+Inspection of the variational formulation tells us that the space $V(\Omega)$
+can be taken to be $H^1_0(\Omega)$. The space $Q(\Gamma)$, in the co-dimension
+zero case, should be taken as $H^1(\Gamma)$, while in the co-dimension one case
+should be taken as $H^{1/2}(\Gamma)$.
+
+The function $g$, therefore, should be either in $H^1(\Gamma)$ (for the
+co-dimension zero case) or $H^{1/2}(\Gamma)$ (for the co-dimension one case).
+This leaves us with a Lagrange multiplier $\lambda$ in $Q^*(\Gamma)$, which is
+either $(H^1(\Gamma))^*$ or $H^{-1/2}(\Gamma)$.
```

Comment:
Looking at how you've written $Q^*(\Gamma)$, should $(H^1(\Gamma))^*$ -> $(H^1)^*(\Gamma)$?